nautobot/extras/tasks.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib.contenttypes.models import ContentType
+from django_rq import job
+
+from nautobot.extras.choices import CustomFieldTypeChoices
+
+
+@job("custom_fields")
+def update_custom_field_choice_data(field_id, old_value, new_value):
+    """
+    Update the values for a custom field choice used in objects' custom_field_data for the given field.
+
+    Args:
+        field_id (uuid4): The PK of the custom field to which this choice value relates
+        old_value (str): The existing value of the choice
+        new_value (str): The value which will be used as replacement
+    """
+    from nautobot.extras.models import CustomField
+
+    field = CustomField.objects.get(pk=field_id)
+
+    if field.type == CustomFieldTypeChoices.TYPE_SELECT:
+        # Loop through all field content types and search for values to update
+        for ct in field.content_types.all():
+            model = ct.model_class()
+            for obj in model.objects.filter(**{f"custom_field_data__{field.name}": old_value}):
+                obj.custom_field_data[field.name] = new_value
+                obj.save()
+
+    elif field.type == CustomFieldTypeChoices.TYPE_MULTISELECT:
+        # Loop through all field content types and search for values to update
+        for ct in field.content_types.all():
+            model = ct.model_class()
+            for obj in model.objects.filter(**{f"custom_field_data__{field.name}__contains": old_value}):
+                old_list = obj.custom_field_data[field.name]
+                new_list = [new_value if e == old_value else e for e in old_list]
+                obj.custom_field_data[field.name] = new_list
+                obj.save()
+
```

Comment:
A couple of thoughts here:

- Would it make sense to create a `JobResult` for either of these new tasks?
- Should this task log an error or failure of some sort if the `field` is not found or if the `field.type` is not in `(TYPE_SELECT, TYPE_MULTISELECT)`?

Reply:
To the first point, I think that is a more fundamental question we need to address in the core. I see these as internal application jobs and thus exposing JobResults that get mixed with userspace jobs would be confusing to end-users. These jobs can still be monitored in the RQ queue and this would be the same in a celery world. Note that the release check job follows this same pattern today.